src/user_interface.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+from src.message import Message as message
+
+
+class UserInterface:
+    def get_user_input(self):
+        user_input = input()
+        is_integer = self.determine_is_integer(user_input)
+        if is_integer:
+            user_integer = int(user_input)
+            return self.validate_move(user_integer)
+        else:
+            return self.handle_incorrect_input(user_input)
+
+    def validate_move(self, user_input):
+        if self.input_in_range(user_input):
+            return user_input
+        else:
+            return self.handle_incorrect_input(user_input)
+
+    def input_in_range(self, user_input):
+        if user_input in range(1, 10):
+            return True
+        else:
+            return False
+
+    def determine_is_integer(self, user_input):
+        if re.match(r"^[0-9]*$", user_input) is None:
```

Comment:
Looks like this is doing a little more than the name suggests--determining specifically an integer in the range. A few ideas that I'd be interested in your thoughts on:

Simply renaming to `determine_is_valid_integer` or something.

Or `determine_is_integer_in_range(self, user_input, range)`.

Either of these seem like a good idea? Which one do you like more and why?

Reply:
LOL I wrote that and then immediately saw your `input_in_range` method.

Reply:
I may have just mistaken the regex to be checking specifically 1-9 vs any integer. I always forget that stuff.

Reply:
that regex is for a string of numbers and only numbers of infinite length.